### PR TITLE
RHDEVDOCS-5565: Expose TEKTON_HUB_API variable in the tektonconfig.

### PR DESCRIPTION
**Pull request by eromanova97**
• **Aligned team**: Dev Tools
• **OCP version for cherry-picking**: enterprise-4.12 and later
• **JIRA issue**:  [RHDEVDOCS-5565](https://issues.redhat.com/browse/RHDEVDOCS-5565)
• **Preview page**: [Configuring the hub resolver](https://63192--docspreview.netlify.app/openshift-enterprise/latest/cicd/pipelines/remote-pipelines-tasks-resolvers#resolver-hub-config_remote-pipelines-tasks-resolvers)
• **SME Review**: Completed by @jkandasa 
• **QE review**: Completed by @ppitonak 
• **Peer-review**: Completed by @mramendi

### CODE REVIEW

**Review by ppitonak — APPROVED** [review]

File: modules/op-resolver-hub-config.adoc [hunk 1]
```diff
@@ -33,9 +33,20 @@ spec:
...
 <5> The default hub for pulling a resource, either `artifact` for {artifact-hub} or `tekton` for {tekton-hub}.
+<6> The {tekton-hub} API used, if the `default-type` option is set to `tekton`.
+<7> The {artifact-hub} API used, if the `default-type` option is set to `artifact`.
++
+[IMPORTANT]
+====
+If you set the `default-type` option to `tekton`, you must configure your own instance of the {tekton-hub} by setting the `tekton-hub-api` value.
+
+If you set the `default-type` option to `artifact` then the resolver uses the public hub API at https://artifacthub.io/ by default. You can configure your own {artifact-hub} API by setting the `artifact-hub-api` value.
+====
```
Comment (jkandasa) [inline modules/op-resolver-hub-config.adoc:52]: @vdemeester I remember we do not support `artifact` in RH Openshift Pipelines. I think in that case we should not include  `artifact` configuration in our documentation.
WDYT?

Comment (vdemeester) [reply modules/op-resolver-hub-config.adoc:52]: We do not ship a supported artifacthub product yes, but we can still document that they can do, don't we ?

Comment (ekristova) [reply modules/op-resolver-hub-config.adoc:52]: @mramendi WDYT?

Comment (jkandasa) [reply modules/op-resolver-hub-config.adoc:52]: @vdemeester I think, we have to mention exclusively that, we support only the configuration part of `artifact` and we do not support `artifact` product.

**Review by jkandasa — APPROVED** [review]
LGTM

CC @vdemeester

File: modules/op-resolver-hub-config.adoc (hunk 1 above)
Comment (mramendi) [reply modules/op-resolver-hub-config.adoc:52]: I agree with @jkandasa - let's document the setting, but for Artifact Hub, put in a note that Artifact Hub custom instances are not currently supported by Red Hat. And we should put a link to "Using Tekton Hub with Pipelines" in the Additional Resources, that's where we document setting up a custom Tekton Hub instance.

File: cicd/pipelines/remote-pipelines-tasks-resolvers.adoc [hunk 2]
```diff
@@ -23,6 +23,11 @@ Git resolver:: Retrieves a task or pipeline binding from a Git repository. You m
 == Specifying a remote pipeline or task from a Tekton catalog
 You can specify a remote pipeline or task that is defined in a public Tekton catalog, either link:https://artifacthub.io/[{artifact-hub}] or link:https://hub.tekton.dev/[{tekton-hub}], by using the hub resolver.
 
+[IMPORTANT]
+====
+The {artifact-hub} product is not supported with {pipelines-title}. Only the configuration of {artifact-hub} is supported.
+====
```
Comment (mramendi) [inline cicd/pipelines/remote-pipelines-tasks-resolvers.adoc:29]: I don't think Artifact Hub is a product to start with. Do you would want to say something like "The tasks and pipelines available on Artifact Hub are not supported..."? Or do you want to say "Deploying a custom instance of Artifact Hub is not supported..."?

File: modules/op-resolver-hub-config.adoc [hunk 3]
```diff
@@ -33,9 +33,20 @@ spec:
...
 <2> The default {artifact-hub} catalog for pulling a task resource.
 <3> The default {artifact-hub} catalog for pulling a pipeline resource.
 <4> The default object kind for references.
 <5> The default hub for pulling a resource, either `artifact` for {artifact-hub} or `tekton` for {tekton-hub}.
+<6> The {tekton-hub} API used, if the `default-type` option is set to `tekton`.
+<7> The {artifact-hub} API used, if the `default-type` option is set to `artifact`.
++
+[IMPORTANT]
+====
+If you set the `default-type` option to `tekton`, you must configure your own instance of the {tekton-hub} by setting the `tekton-hub-api` value.
```
Comment (mramendi) [inline modules/op-resolver-hub-config.adoc:49]: Is this actually true? Isn't there a default Tekton Hub?

File: modules/op-resolver-hub-config.adoc [hunk 4]
```diff
@@ -33,9 +33,20 @@ spec:
...
       default-type: tekton # <5>
+      tekton-hub-api: "https://my-custom-tekton-hub.example.com" # <6>
+      artifact-hub-api: "https://my-custom-artifact-hub.example.com" # <7>
 ----
 <1> The default {tekton-hub} catalog for pulling a resource.
 <2> The default {artifact-hub} catalog for pulling a task resource.
 <3> The default {artifact-hub} catalog for pulling a pipeline resource.
 <4> The default object kind for references.
 <5> The default hub for pulling a resource, either `artifact` for {artifact-hub} or `tekton` for {tekton-hub}.
+<6> The {tekton-hub} API used, if the `default-type` option is set to `tekton`.
```
Comment (mramendi) [inline modules/op-resolver-hub-config.adoc:44]: I think this one and the next one should be marked as optional more clearly.

File: modules/op-resolver-hub-config.adoc (hunk 3 above)
Comment (ekristova) [reply modules/op-resolver-hub-config.adoc:49]: Hello @jkandasa can you please confirm the above? Thank you!

Comment (jkandasa) [reply modules/op-resolver-hub-config.adoc:49]: @ekristova There is no public(or default) tekton hub. So we must to set it manually.
Source: https://github.com/tektoncd/pipeline/blob/main/docs/hub-resolver.md#configuring-the-hub-api-endpoint
```
When setting the type field to tekton, you must configure your own instance of the Tekton Hub by setting the TEKTON_HUB_API environment variable...
```

File: cicd/pipelines/remote-pipelines-tasks-resolvers.adoc (hunk 2 above)
Comment (jkandasa) [reply cicd/pipelines/remote-pipelines-tasks-resolvers.adoc:29]: @mramendi we want to support Artifact as hub resolver in task or pipelines.
But we will not support, if there is an issue reported by customer in Artifact Hub product itself.

cc @vdemeester

Comment (mramendi) [reply cicd/pipelines/remote-pipelines-tasks-resolvers.adoc:29]: @jkandasa @vdemeester I am not sure how Artifact Hub can be described as a product

File: modules/op-resolver-hub-config.adoc (hunk 4 above)
Comment (ekristova) [reply modules/op-resolver-hub-config.adoc:44]: Hello @mramendi I will wait for the answer from SME regarding comment above about the Tekton Hub API value. But the information about having to set the `tekton-hub-api` value is in the supporting docs. If that will be confirmed then the `tekton-hub-api` value is not optional.

File: cicd/pipelines/remote-pipelines-tasks-resolvers.adoc (hunk 2 above)
Comment (jkandasa) [reply cicd/pipelines/remote-pipelines-tasks-resolvers.adoc:29]: @mramendi Artifact Hub maintained out side of tekton community.
and It is a web-based application. So I called it as a product.
* https://artifacthub.io/
* https://artifacthub.io/docs/

May be the `project` is the right word?
You can rename it as required. Thanks @mramendi

File: modules/op-resolver-hub-config.adoc (hunk 4 above)
Comment (ekristova) [reply modules/op-resolver-hub-config.adoc:44]: Note to the previous comment: I had some a minor issue with GitHub and did not see the responses right away, but the response is already there, therefore the the `tekton-hub-api` value is not optional.

Comment (mramendi) [reply modules/op-resolver-hub-config.adoc:44]: Agreed - I did not know this at the time I wrote the comment. "Optional" applies only to the artifact hub setting (which in fact is normally *not* needed)

**Review by mramendi — APPROVED** [review]

**Comment by bburt-rh** [comment]
Merged into main, and CP'ed to 4.12, 4.13, and 4.14 branches.

**Pull request merged**